exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporter.java

```diff
@@ -0,0 +1,225 @@
+      requestBuilder.headers(headers);
+    }
+    RequestBody requestBody =
+        RequestBody.create(
+            exportTraceServiceRequest.toByteArray(), MediaType.parse("application/x-protobuf"));
+    if (isCompressionEnabled) {
+      requestBuilder.addHeader("Content-Encoding", "gzip");
+      requestBuilder.post(gzipRequestBody(requestBody));
+    } else {
+      requestBuilder.post(requestBody);
+    }
+
+    CompletableResultCode result = new CompletableResultCode();
+
+    client
+        .newCall(requestBuilder.build())
+        .enqueue(
+            new Callback() {
+              @Override
+              public void onFailure(Call call, IOException e) {
+                spansExportedFailure.add(spans.size());
+                result.fail();
+                logger.log(
+                    Level.SEVERE,
+                    "Failed to export spans. The request could not be executed. Full error message: "
+                        + e.getMessage());
+              }
+
+              @Override
+              public void onResponse(Call call, Response response) {
+                if (response.isSuccessful()) {
+                  spansExportedSuccess.add(spans.size());
+                  result.succeed();
+                  return;
+                }
+
+                spansExportedFailure.add(spans.size());
+                int code = response.code();
+
+                Status status = extractErrorStatus(response);
+
+                logger.log(
+                    Level.WARNING,
+                    "Failed to export spans. Server responded with HTTP status code "
+                        + code
+                        + ". Error message: "
+                        + status.getMessage());
+                result.fail();
+              }
+            });
+
+    return result;
+  }
+
+  private static RequestBody gzipRequestBody(RequestBody requestBody) {
+    return new RequestBody() {
+      @Override
+      public MediaType contentType() {
+        return requestBody.contentType();
+      }
+
+      @Override
+      public long contentLength() {
+        return -1;
+      }
+
+      @Override
+      public void writeTo(BufferedSink bufferedSink) throws IOException {
+        BufferedSink gzipSink = Okio.buffer(new GzipSink(bufferedSink));
+        requestBody.writeTo(gzipSink);
+        gzipSink.close();
+      }
+    };
+  }
+
+  private static Status extractErrorStatus(Response response) {
+    try {
+      ResponseBody responseBody = response.body();
+      if (responseBody == null) {
+        return Status.newBuilder()
+            .setMessage("Unable to extract error message from empty response body.")
+            .setCode(Code.UNKNOWN.getNumber())
+            .build();
+      } else {
+        return Status.parseFrom(responseBody.bytes());
+      }
+    } catch (IOException e) {
+      return Status.newBuilder()
+          .setMessage("Unable to extract error message from response: " + e.getMessage())
```

Comment:
In the context of my above comment, I think the response.getMessage() is fine for this and we can ignore e.getMessage() which is unlikely to be helpful for responses from off-the-shelf proxies.

Reply:
Ok. I do think it will be helpful to have the status message be different based on whether the response body is empty, or present but failed to parse. Definitely agree that the status message is more likely to contain useful information than `e.getMessage()`.